### PR TITLE
feat(queries): add terraform nifcloud queries

### PR DESCRIPTION
**Pull request by tunakyonn**
**Proposed Changes**
-  NIFCLOUD is a famous cloud in Japan! Added these Terraform NIFCLOUD queries.
    - https://registry.terraform.io/providers/nifcloud/nifcloud/latest
- And added Terraform NIFCLOUD queries docs.
    - Fix `make generate-queries-docs` and execute!

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -148,12 +148,12 @@ run-local: build
 .PHONY: generate-queries-docs
 generate-queries-docs: ## generate queries catalog md files
 	$(call print-target)
-	@pip3 install -r .github/generators/requirements.txt
-	@python3 -u .github/generators/docs_generator.py \
+	@pip3 install -r .github/scripts/docs-generator/requirements.txt
+	@python3 -u .github/scripts/docs-generator/docs_generator.py \
```
Comment (gabriel-cx) [inline Makefile:152]: kindly use the following suggestion:

```suggestion
	@python3 -u .github/scripts/docs-generator/docs-generator.py \
```

**Comment by gabriel-cx** [comment]
Hi @tunakyonn,

Thank you so much for this amazing contribution! 😄
We already did a first level review and we added a suggestion of code change.
After you submit the suggested change, we will perform the final and complete review ASAP.

Thank you again!

File: Makefile (hunk 1 above)
Comment (tunakyonn) [reply Makefile]: Thank you. Fixed！

**Comment by tunakyonn** [comment]
Hi @gabriel-cx ,

Thank you for your review! :smiley:
Fixed errors in comments and tests.
Please review again.

**Comment by gabriel-cx** [comment]
Hi @tunakyonn ,

Thank you, the changes you did recently looks good! 👍 
In order to add a new cloud provider, you will need to change a bit more files.
You can check this [discussion here](https://github.com/Checkmarx/kics/discussions/6088), there you can find the information you need!

Feel free to leave us a message anytime! 😄

**Comment by tunakyonn** [comment]
Hi @gabriel-cx ,

Thank you for your advice! 😃
I fixed the file with reference to the p-r I was taught.
So please review again.

**Comment by gabriel-cx** [comment]
hi @tunakyonn ,
The code looks cool! thank you! 😃
Now, we will inform our AppSec team to review your queries. After their approval, we can merge your PR!
Kindly be aware that this procedure can take some time.

Note: Looks like some unit-tests are failing due to duplicated descriptions ids.
image

**Comment by gabriel-cx** [comment]
hi @tunakyonn,

Thanks for providing your thoughts on this! I will recommend the following:

1. nifcloud engine support
    - pkg/engine/source/filesystem.go
    - pkg/engine/source/filesystem_test.go
    - .github/scripts/queries-validator/metadata-schema.json
    - test/main_test.go
    - internal/constants/constants.go
    - docs/*

2. nifcloud queries per platform / provider (so it would be more manageable for review) + a meta issue to track them
    - assets/queries/terraform/nifcloud/*

3. changes / fixes unrelated to nifcloud (e.g. the makefile change).
    - Makefile

**Pull request closed without merging**